agent/agent.go

```diff
@@ -88,19 +89,24 @@ func _main() int {
 		log.Warn("SSL certificate verification disabled. This is not recommended.")
 	}
 	log.Info("Loading configuration")
-	cfg, err := config.NewConfig(ec2MetadataClient)
-	// Load cfg before doing 'versionFlag' so that it has the DOCKER_HOST
-	// variable loaded if needed
+	cfg, cfgErr := config.NewConfig(ec2MetadataClient)
+	// Load cfg and create Docker client before doing 'versionFlag' so that it has the DOCKER_HOST variable loaded if needed
+	clientFactory := dockerclient.NewFactory(cfg.DockerEndpoint)
+	dockerClient, err := engine.NewDockerGoClient(clientFactory, cfg.EngineAuthType, cfg.EngineAuthData, *acceptInsecureCert)
+	if err != nil {
+		log.Criticalf("Error creating Docker client: %v", err)
+		return exitcodes.ExitError
+	}
 	if *versionFlag {
-		versionableEngine := engine.NewTaskEngine(cfg, *acceptInsecureCert)
+		versionableEngine := engine.NewTaskEngine(cfg, dockerClient)
 		version.PrintVersion(versionableEngine)
 		return exitcodes.ExitSuccess
 	}
 
 	sighandlers.StartDebugHandler()
 	ctx := context.Background()
 
-	if err != nil {
+	if cfgErr != nil {
```

Comment:
It's intentionally farther down so that errors from reading config don't prevent the version command line flag from working.